dask/diagnostics/progress.py

```diff
@@ -37,6 +37,8 @@ class ProgressBar(Callback):
         Width of the bar
     dt : float, optional
         Update resolution in seconds, default is 0.1 seconds
+    out : file object, optional
+        File object to which the progress bar will be written. Default is sys.stdout.
```

Comment:
I'm not sure if this explanation makes the behavior clearer, if you look at the original issue, passing a file object does not always work. The `out` parameter is really intended as a place where you can set `sys.stdout` or `sys.stderr`.

Reply:
Oops! I just saw that Ian has much more thorough feedback! Feel free to ignore this :point_up:

Reply:
> Just FYI it has to be a code-owner who comments "ok to test" for GPU CI to start running, but I like your initiative @ppsbs 😄

that's so embarassing! Thanks for the tip @jsignell! :D

Reply:
As @ian-r-rose , mentioned, out expects a file object in 'w' mode and TemporaryFile defaults to 'w+b' (https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryFile). But it's also true that, as you mentioned, from the previous modifications, it looks like only sys.stdout or sys.stderr were supposed to be used. 

I will think about it and update the fix. Thank you for your valuable tips @jsignell and @ian-r-rose !